packages/core/src/connection/transactional-connection.ts

```diff
@@ -78,14 +79,28 @@ export class TransactionalConnection {
                 return transactionManager.getRepository(maybeTarget!);
             } else {
                 // tslint:disable-next-line:no-non-null-assertion
-                return getRepository(maybeTarget!);
+                return this.getOuterRepository(maybeTarget!);
             }
         } else {
             // tslint:disable-next-line:no-non-null-assertion
-            return getRepository(ctxOrTarget ?? maybeTarget!);
+            return this.getOuterRepository(ctxOrTarget ?? maybeTarget!);
         }
     }
 
+    /**
+     * @description
+     * Returns a TypeORM repository, which is not aware of any existing transaction. 
+     * Therefore calling this method is discouraged without a deliberate reason.
+     * It's recommended to use {@link TransactionalConnection.getRepository getRepository(ctx, Entity)} instead.
+     * 
+     * @since 1.6.3
+     */
+    getOuterRepository<Entity>(
```

Comment:
I'm not 100% sure I like this naming. Will people know what is meant by "outer repository"? I wonder if there could be a name which better communicates what this is doing. 

- `getRepositoryOutsideTransaction`?
- `getRepositoryWithoutContext`?

not sure I like those any better... Any other ideas?

Reply:
Well, `getNoContextRepository()` is an option, but It's not better than you said. 
Another option is to delete getOuterRepository function completely, and instead force developer to use `get() rawConnection`. For example: `this.connection.rawConnection.getRepository()`. By the way, this class is called `TransactionalConnection` which means that every member of this function, like `getRepository()` or smth else should relate somehow to transaction. If user does not need transactional repository we may force developer to inject original typeorm Connection class.